### DISCUSSION TITLE
problem with accessing pdf file

### DISCUSSION THREAD

**Discussion by johaankjis** [beeware/toga#2864] [Debugging help]
i wanted to access a pdf through beeware and i came across fitz library to be used. i have done the code changes but when running the program on briefcase dev, it open windows saying that the file is not found. how to fix it?

**Comment by mhsmith** [comment]
Please post:
* The full error message (as text if possible, otherwise as a screenshot).
* The relevant sections of your code.

**Reply by johaankjis** [reply]
I have attached the code along with this reply. File not found: src/fondren_library_map.pdf. Here is the code of the app.py
```python
import toga
from toga.style import Pack
from toga.style.pack import COLUMN
import fitz  # PyMuPDF
import os


class FondrenApp(toga.App):

    def startup(self):
        # Print the current working directory
        print("Current working directory:", os.getcwd())

        # Create main box layout
        main_box = toga.Box(style=Pack(direction=COLUMN, padding=10))

        # Load and display the PDF
        self.pdf_path = "src/fondren_library_map.pdf"

        # Check if the PDF file exists
        if not os.path.exists(self.pdf_path):
            print(f"File not found: {self.pdf_path}")
            main_box.add(toga.Label("PDF file not found!", style=Pack(padding=10)))
        else:
            print(f"File found: {self.pdf_path}")
            self.pages = self.render_pdf_pages(self.pdf_path)

            # Add images (rendered PDF pages) to the main box
            for page in self.pages:
                img = toga.ImageView(page, style=Pack(height=300, width=300))  # Adjust size
                main_box.add(img)

        # Create a main window and add the box
        self.main_window = toga.MainWindow(title=self.name)
        self.main_window.content = main_box
        self.main_window.show()

    def render_pdf_pages(self, pdf_file):
        doc = fitz.open(pdf_file)
        page_images = []

        for page_num in range(len(doc)):
            page = doc.load_page(page_num)  # Load the page
            pix = page.get_pixmap()  # Get page as an image
            image_path = f"src/page_{page_num}.png"
            pix.save(image_path)  # Save as PNG
            page_images.append(toga.Image(image_path))

        return page_images


def main():
    return FondrenApp("Fondren Library Navigation", "com.example.fondren")


if __name__ == '__main__':
    main().main_loop()

```

**Reply by freakboy3742** [reply]
The problem is with this line of code:
```
  self.pdf_path = "src/fondren_library_map.pdf"
```

This is specifying a relative path - it doesn't specify a complete path on the filesystem. Whenever you specify a relative path, the operating system will use the current working directory as the base from which to convert the relative path into an absolute path - but due to differences in how apps are packaged, the relationship between the current working directory and your data file won't be necessarily be consistent.

So - you need to use an absolute URL. That means either specifying a `/full/path/to/your/src/file.pdf`, or, building the path relative to some reliable location. Toga provides the `app.paths` attribute to provide some anchors that act as known locations in the filesystem - so, for example `self.paths.app / "file.pdf"` would be a `file.pdf` in the same folder as the app.py that defines your app. There are some other options on the `paths` object; [see the Toga documentation for details](https://toga.readthedocs.io/en/stable/reference/api/resources/app_paths.html#toga.paths.Paths).

**Reply by johaankjis** [reply]
I see. I just implemented the changed. It is now working. Thanks a lot!
